js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+            //     /trades, /historicalTrades
+            //
+            //     [
+            //         {
+            //             "id": null,
+            //             "price": "40798.94",
+            //             "qty": "0.000508",
+            //             "quoteQty": "20.72586152",
+            //             "time": "1647546934374",
+            //             "isBuyerMaker": true,
+            //             "isBestMatch": true
+            //         },
+            //     ]
+            //
+            //     /aggrTrades
+            //
+            //     [
+            //         {
+            //           "a": null,
+            //           "f": null,
+            //           "l": null,
+            //           "p": "40679",
+            //           "q": "0.001309",
+            //           "T": 1647551328000,
+            //           "m": true,
+            //           "M": true
+            //         },
+            //     ]
+            //
+        } else if (market['swap']) {
+            const response = await this.contractPublicGetDealsSymbol (this.extend (request, params));
+            //
+            //     {
+            //         "success": true,
+            //         "code": 0,
+            //         "data": [
+            //             {
+            //                 "p": 31199,
+            //                 "v": 18,
+            //                 "T": 1,
+            //                 "O": 3,
+            //                 "M": 2,
+            //                 "t": 1609831235985
+            //             },
+            //         ]
+            //     }
+            //
+            trades = this.safeValue (response, 'data');
+        }
+        return this.parseTrades (trades, market, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        let id = undefined;
+        let timestamp = undefined;
+        let orderId = undefined;
+        let symbol = undefined;
+        let fee = undefined;
+        const type = undefined;
+        let side = undefined;
+        let takerOrMaker = undefined;
+        let priceString = undefined;
+        let amountString = undefined;
+        let costString = undefined;
+        // if swap
+        if ('v' in trade) {
+            //
+            // spot: fetchTrades
+            //
+            //     {
+            //         "p": 31199,
+            //         "v": 18,
+            //         "T": 1,
+            //         "O": 3,
+            //         "M": 2,
+            //         "t": 1609831235985
+            //     }
+            //
+            timestamp = this.safeInteger (trade, 't');
+            market = this.safeMarket (undefined, market);
+            symbol = market['symbol'];
+            priceString = this.safeString (trade, 'p');
+            amountString = this.safeString (trade, 'v');
+            side = this.parseOrderSide (this.safeString (trade, 'T'));
+            takerOrMaker = 'taker';
```

Comment:
Can we be sure that every trade will be a "taker" one?

Reply:
Public trades don't really differentiate between `taker` or `maker`. We can assume that all public trades are takers.

Reply:
@kroitor thanks for the clarification